telescope/src/main/java/com/mattprecious/telescope/TelescopeLayout.java

```diff
@@ -129,13 +153,62 @@ public TelescopeLayout(Context context, AttributeSet attrs, int defStyle) {
     });
 
     if (isInEditMode()) {
+      projectionManager = null;
+      windowManager = null;
       vibrator = null;
       screenshotFolder = null;
+      requestCaptureFilter = null;
+      requestCaptureReceiver = null;
       return;
     }
 
+    windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
     vibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
     screenshotFolder = getScreenshotFolder(context);
+
+    if (SDK_INT < LOLLIPOP) {
+      projectionManager = null;
+      requestCaptureFilter = null;
+      requestCaptureReceiver = null;
+    } else {
+      projectionManager =
+          (MediaProjectionManager) context.getSystemService(Context.MEDIA_PROJECTION_SERVICE);
+
+      requestCaptureFilter =
+          new IntentFilter(RequestCaptureActivity.getResultBroadcastAction(context));
+      requestCaptureReceiver = new BroadcastReceiver() {
+        @TargetApi(Build.VERSION_CODES.LOLLIPOP) @Override
+        public void onReceive(Context context, Intent intent) {
+          int resultCode = intent.getIntExtra(RequestCaptureActivity.RESULT_EXTRA_CODE,
+              Activity.RESULT_CANCELED);
+          Intent data = intent.getParcelableExtra(RequestCaptureActivity.RESULT_EXTRA_DATA);
+
+          final MediaProjection mediaProjection =
+              projectionManager.getMediaProjection(resultCode, data);
+          if (mediaProjection == null) {
+            captureCanvasScreenshot();
+            return;
+          }
+
+          // Delay capture until after the permission dialog is gone.
+          postDelayed(new Runnable() {
+            @Override public void run() {
+              captureNativeScreenshot(mediaProjection);
+            }
+          }, 1000);
+        }
+      };
+    }
+  }
+
+  @Override protected void onAttachedToWindow() {
+    super.onAttachedToWindow();
+    registerRequestCaptureReceiver();
+  }
+
+  @Override protected void onDetachedFromWindow() {
+    super.onDetachedFromWindow();
+    unregisterRequestCaptureReceiver();
```

Comment:
If the user has multiple activities this will stay registered when they navigate to them. The view never detached from the hierarchy despite not being visible on screen. I think you'll want to register immediately before starting the second activity and then unregister immediately in the callback.